### DISCUSSION TITLE
Register Plugins In a Loop

### DISCUSSION THREAD

**Discussion by elverskog** [fastify/fastify#4442] [Q&A] [answered]
I'm trying to register plugins in a loop.

The use case:
I have a environment that includes vite and marko.
Each route is declared in various files in /src/pages.
It is based off this example:
https://github.com/marko-js/examples/tree/master/examples/vite-fastify

Rather than having a  line for each route declaration(s) file like:
`fastify.register(import("./pages/home/home.page"));`

I want to have a config object with all my pages, that includes, not only said file path but also the pathname itself, access required etc.
Like this:
```
{
      upload: {
        pathName: "/upload",
        filePath: "./pages/upload/upload.page",
        userTypeRequired: "any",
        noAccessNotice: "Please log in to upload a file."
      }, 
  ...
  }
```
I then want to loop through the keys/objects in that object.
```
Object.keys(routes).forEach(key => {
  app.register(import(/* @vite-ignore */routes[key].filePath));
});
```
This works when I run it in dev mode
But fails when I try to build, I get a vite error:
_[marko-vite:post] No Marko files were found when bundling the server in linked mode._

What seems to be happening here is that no routes and hence Marko templates are found.

I'm hoping there is some way to declare these routes dynamically

**Comment by L2jLiga** [comment, marked as answer]
Hi there, did you tried to use import instead of filePath in your config object? 
I'm not familiar with Marko (and/or vite) but I can guess it can't resolve routes because they're not explicitly imported

**Comment by elverskog** [comment]
Hey. Thanks for getting back to me so soon.

Do you mean I should try importing in the config object?
So that the register() doesn't have a import  as an argument but rather an object (assigned to a key in the object)?
Regardless, I will try that, of course

Per "routes because they're not explicitly imported"
I was about to add the original post when I saw your reply...

The build error definitely had nothing to do with the loop but rather the dynamic file path 
Vite just seems to want a string in the import
routes.home.pathName fails

I guess it breaks when trying to "scan" the file; validating that the paths are legit

**Reply by L2jLiga** [reply]
My guesses are based on experience with webpack, it's unable to import with dynamic path (or at least wasn't able before), and the only solution is to use normal imports

**Reply by elverskog** [reply]
@L2jLiga Just curious, by normal imports your mean the path as a string as oppose to a variable?

That said, moving the imports into the config object, as suggested/described worked
Ty!

**Reply by L2jLiga** [reply]
> Just curious, by normal imports your mean the path as a string as oppose to a variable?

Yes, `import('./file-a.js')` instead of `import(path)`
